src/ir_Argo.cpp

```diff
@@ -0,0 +1,282 @@
+  checksum();
+}
+
+bool IRArgoESP::getMax() {
+  return max_mode;
+}
+
+// Set the temp in deg C
+// Sending 0 equals +4
+void IRArgoESP::setTemp(uint8_t temp) {
+  if (temp < ARGO_MIN_TEMP)
+    temp = ARGO_MIN_TEMP;
+  else if (temp > ARGO_MAX_TEMP)
+    temp = ARGO_MAX_TEMP;
+
+  // Store in attributes
+  set_temp = temp;
+  // offset 4 degrees. "If I want 12 degrees, I need to send 8"
+  temp -= 4;
+  // Settemp = Bit 6,7 of byte 2, and bit 0-2 of byte 3
+  // mask out bits
+  // argo[13] & 0x00000100;  // mask out ON/OFF Bit
+  argo[2] &= 0b00111111;
+  argo[3] &= 0b11111000;
+
+  argo[2] += temp << 6;  // append to bit 6,7
+  argo[3] += temp >> 2;  // remove lowest to bits and append in 0-2
+  checksum();
+}
+
+uint8_t IRArgoESP::getTemp() {
+  return set_temp;
+}
+
+// Set the speed of the fan
+void IRArgoESP::setFan(uint8_t fan) {
+  // Set the fan speed bits, leave low 4 bits alone
+  fan_mode = fan;
+  // Mask out bits
+  argo[3] &= 0b11100111;
+  // Set fan mode at bit positions
+  argo[3] += fan << 3;
+  checksum();
+}
+
+uint8_t IRArgoESP::getFan() {
+  return fan_mode;
+}
+
+void IRArgoESP::setFlap(uint8_t flap) {
+  flap_mode = flap;
+  // TODO(kaschmo):
+  checksum();
+}
+
+uint8_t IRArgoESP::getFlap() {
+  return flap_mode;
+}
+
+uint8_t IRArgoESP::getMode() {
+  // return cooling 0, heating 1
+  return ac_mode;
+}
+
+void IRArgoESP::setCoolMode(uint8_t mode) {
+  ac_mode = 0;  // Set ac mode to cooling
+  cool_mode = mode;
+  // Mask out bits, also leave bit 5 on 0 for cooling
+  argo[2] &= 0b11000111;
+
+  // Set cool mode at bit positions
+  argo[2] += mode << 3;
+  checksum();
+}
+
+uint8_t IRArgoESP::getCoolMode() {
+  return cool_mode;
+}
+
+void IRArgoESP::setHeatMode(uint8_t mode) {
+  ac_mode = 1;  // Set ac mode to heating
+  heat_mode = mode;
+  // Mask out bits
+  argo[2] &= 0b11000111;
+  // Set heating bit
+  argo[2] |= 0b00100000;
+  // Set cool mode at bit positions
+  argo[2] += mode << 3;
+  checksum();
+}
+
+uint8_t IRArgoESP::getHeatMode() {
+  return heat_mode;
+}
+
+void IRArgoESP::setNight(bool state) {
+  night_mode = state;
+  if (night_mode)
+    // Set bit at night position: bit 2
+    argo[9] |= 0b00000100;
+  else
+    argo[9] &= 0b11111011;
+  checksum();
+}
+
+bool IRArgoESP::getNight() {
+  return night_mode;
+}
+
+void IRArgoESP::setiFeel(bool state) {
+  ifeel_mode = state;
+  if (ifeel_mode)
+    // Set bit at iFeel position: bit 7
+    argo[9] |= 0b10000000;
+  else
+    argo[9] &= 0b01111111;
+  checksum();
+}
+
+bool IRArgoESP::getiFeel() {
+  return ifeel_mode;
+}
+
+void IRArgoESP::setTime() {
+  // TODO(kaschmo): use function call from checksum to set time first
+}
+
+void IRArgoESP::setRoomTemp(uint8_t temp) {
+  // Use function call from checksum to set room temp
+  temp -= 4;
+  // Mask out bits
+  argo[3] &= 0b00011111;
+  argo[4] &= 0b11111100;
+
+  argo[3] += temp << 5;  // Append to bit 5,6,7
+  argo[4] += temp >> 3;  // Remove lowest 3 bits and append in 0,1
+  checksum();
```

Comment:
If you add this to where the other comments say, you can remove this instance of `checksum()`.